akka-testkit-typed/src/main/scala/akka/typed/testkit/StubbedActorContext.scala

```diff
@@ -1,14 +1,108 @@
 package akka.typed.testkit
 
+import akka.actor.InvalidMessageException
 import akka.{ actor ⇒ untyped }
 import akka.actor.typed._
 import akka.util.Helpers
+import akka.{ actor ⇒ a }
+import akka.util.Unsafe.{ instance ⇒ unsafe }
 
 import scala.collection.immutable.TreeMap
 import scala.concurrent.ExecutionContextExecutor
 import scala.concurrent.duration.FiniteDuration
 import akka.annotation.InternalApi
-import akka.actor.typed.internal.ActorContextImpl
+import akka.actor.typed.internal.{ ActorContextImpl, ActorRefImpl }
+
+import scala.annotation.tailrec
+import scala.util.control.NonFatal
+
+/**
+ * A local synchronous ActorRef that invokes the given function for every message send.
+ * This reference can be watched and will do the right thing when it receives a [[akka.actor.typed.internal.DeathWatchNotification]].
+ * This reference cannot watch other references.
+ */
+private[akka] final class FunctionRef[-T](
```

Comment:
Why did this end up in the testkit?

Reply:
It used to be used in the old impl, now it is only used in the stubbed actor context. I guess we can re-evaluate how that works during the testkit issues for typed.

Reply:
okok